src/clr-angular/emphasis/alert/providers/multi-alert.service.ts

```diff
@@ -57,6 +57,9 @@ export class MultiAlertService {
     }
 
     previous() {
+        if (this.activeAlerts.length === 0) {
```

Comment:
The concern I see here is that we don't set the `current` property back to 0 when we close. So `current` could remain at a higher index, and when a new alert is added it would be out of bounds, if I'm reading it correctly here.

Reply:
The only way we should be able to enter this block is if current already equals 0.  The service initializes at 0.  Once some number of alert instances are added, the only way for the length to get back to 0 is if each alert instance is closed.  When the last remaining alert instance is there, current will be 0.  Once it is closed, the block catches, and does not alter current, so it remains at 0, ready for any future alert instances that may be added.